start-server-and-test.js

```diff
@@ -0,0 +1,102 @@
+const { spawn } = require('child_process');
+const fetch = require('node-fetch');
+
+/* eslint-disable no-console */
+
+const serverUrl = 'http://localhost:4000';
+const serverTimeout = 360000; // 6 minutes
+
+const waitForServer = async (url, timeout = 30000) => {
+  const start = Date.now();
+  while (Date.now() - start < timeout) {
+    try {
+      const response = await fetch(url);
+      if (response.ok) {
+        return;
+      }
+    } catch (err) {
+      console.error('Waiting for server to respond.');
+    }
+    await new Promise(resolve => setTimeout(resolve, 1000));
+  }
+  throw new Error('Server did not start within the timeout period');
+};
+
+const killPortProcess = async port => {
+  try {
+    const fkill = await import('fkill');
+    await fkill(`:${port}`, { force: true });
```

Comment:
I dont think we should kill other user processes, usually if the port is already allocated, then we just throw an error and wait for the user to kill them, but not us. And without this, I think we can get rid of the `fkill` dependency

Reply:
But we'll be informing them that there was another process running on that port before killing it. What prompted me to add this was that when I used to cancel a test execution midway, sometimes the port would still be occupied, causing the next test run to fail.
Isn't having this automated beneficial as opposed to the user having to kill the processes manually, considering the user meant to execute visual tests knowing that it will need port 4000?

Reply:
Hmm so we would probably need to handle the cancel action? and kill the dev server there. Would it be possible? The thing is that if a user has another process unrelated to kds running in that port, we cant just kill it, since it would be a bit invasive

Reply:
I understand that. Will look into it. Maybe giving the user a nice warning and gracefully exiting the testing environment would be better.